jiant/tasks/seq2seq.py

```diff
@@ -0,0 +1,113 @@
+"""Task definitions for sequence-to-sequence tasks."""
+import codecs
+import collections
+import math
+import os
+from typing import Iterable, List, Sequence, Type
+
+from allennlp.data import Instance
+from allennlp.data.token_indexers import SingleIdTokenIndexer
+from allennlp.training.metrics import Average, BooleanAccuracy
+
+from ..utils.data_loaders import process_sentence
+from .registry import register_task
+from .tasks import (
+    UNK_TOK_ALLENNLP,
+    UNK_TOK_ATOMIC,
+    SequenceGenerationTask,
+    atomic_tokenize,
+    sentence_to_text_field,
+)
+
+
+@register_task("seg_wix", rel_path="seg/wix/", max_targ_v_size=200)
+class CharSeq2SeqTask(SequenceGenerationTask):
+    """Character-based Sequence-to-sequence Task"""
+
+    def __init__(self, path, max_seq_len, max_targ_v_size, name, **kw):
+        """ """
+        super().__init__(name, **kw)
+        self.scorer2 = BooleanAccuracy()
+        self.scorers.append(self.scorer2)
+        self.val_metric = "%s_accuracy" % self.name
+        self.val_metric_decreases = False
+        self.max_seq_len = max_seq_len
+        self._label_namespace = self.name + "_tokens"
+        self.max_targ_v_size = max_targ_v_size
+        self.target_indexer = {"words": SingleIdTokenIndexer(namespace=self._label_namespace)}
+        self.files_by_split = {
+            split: os.path.join(path, "%s.tsv" % split) for split in ["train", "val", "test"]
+        }
+
+    def load_data(self):
+        # Data is exposed as iterable: no preloading
+        pass
+
+    def get_split_text(self, split: str):
+        """
+        Get split text as iterable of records.
+        Split should be one of 'train', 'val', or 'test'.
+        """
+        return self.get_data_iter(self.files_by_split[split])
+
+    def get_all_labels(self) -> List[str]:
+        """ Build character vocabulary and return it as a list """
+        word2freq = collections.Counter()
```

Comment:
I think this is right (assuming "SplitChars" is right in `get_data_iter`), but to minimize confusion, `word2freq` -> `char2freq`